pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -324,6 +338,24 @@ func (nc *nodeConfig) waitForNodeAnnotation(annotation string) error {
 	return nil
 }
 
+// configureCNI populates the CNI config template and sends the config file location
+// for completing CNI configuration in the windows VM
+func (nc *nodeConfig) configureCNI() error {
+	// set the hostSubnet value in the clusterServiceCIDR struct
+	nc.nodeCfgOptions.SetHostSubnet(nc.node.Annotations[HybridOverlaySubnet])
+	// populate the CNI config file with the host subnet and the service network CIDR
+	configFile, err := nc.nodeCfgOptions.PopulateCniConfig(nc.clusterServiceCIDR)
+	defer os.RemoveAll(configFile)
```

Comment:
I suggest not deferring this but calling this explicitly at the end and logging errors.